prov/efa/src/rxr/rxr_ep.c

```diff
@@ -506,12 +506,26 @@ struct rxr_tx_entry *rxr_ep_alloc_tx_entry(struct rxr_ep *rxr_ep,
 }
 
 void rxr_inline_mr_reg(struct rxr_domain *rxr_domain,
-		       struct rxr_tx_entry *tx_entry)
+		       struct rxr_tx_entry *tx_entry, bool is_shm)
 {
 	ssize_t ret;
 	size_t offset;
-	int index;
+	int index = 0;
 
+	if (is_shm) {
+		tx_entry->iov_mr_start = index;
+		while (index < tx_entry->iov_count) {
+			ret = fi_mr_reg(&rxr_domain->util_domain.domain_fid,
```

Comment:
I remember our conversation about using a regv() instead, but someone has to have this loop given our mr_Iov_limit, either here or the bottom layer. I wonder if https://github.com/ofiwg/libfabric/pull/5664 makes this better from a code perspective (have not read that PR yet).

Reply:
#5664 right now does not handle the loop. It does ibv_reg_mr, ofi_mr_map_insert, and shm 's fi_mr_reg. 
Since we already had this loop for rdm domain, I just keep the same pattern here. If we decide to lower this loop down to efa_mr_reg*, I will change accordingly. @dipti-kothari @wzamazon please let me know your thoughts.

Reply:
Nah, that's alright, I was just curious. After reading the other PR, I think we can leave this be for now and revist how we do regv after all the squash changes land.

Reply:
Sure.